k8s/src/main/scala/io/buoyant/k8s/EndpointsNamer.scala

```diff
@@ -104,50 +113,49 @@ abstract class EndpointsNamer(
 
   import EndpointsNamer._
 
-  val idLen = 3
   def lookup(path: Path): Activity[NameTree[Name]]
 
   private[k8s] def lookupServices(
+    nsName: String,
+    portName: String,
+    serviceName: String,
     cache: NsCache,
     serviceCache: ServiceCache,
     id: Path,
     residual: Path
-  ): Activity[NameTree[Name]] = id.take(idLen) match {
-    case id@Path.Utf8(nsName, portName, serviceName) =>
-      cache.services.flatMap { services =>
-        log.debug("k8s ns %s initial state: %s", nsName, services.keys.mkString(", "))
-        services.get(serviceName) match {
+  ): Activity[NameTree[Name]] = cache.services.flatMap { services =>
```

Comment:
Did anything change in this method other than taking the ns, port, and service names from method parameters instead of extracting them from the id?  The diff is a bit tough to read because it looks like some cases got re-ordered.